appscale/tools/agents/ec2_agent.py

```diff
@@ -405,17 +405,20 @@ def run_instances(self, count, parameters, security_configured, public_ip_needed
     instance_type = parameters[self.PARAM_INSTANCE_TYPE]
     keyname = parameters[self.PARAM_KEYNAME]
     group = parameters[self.PARAM_GROUP]
-    spot = parameters[self.PARAM_SPOT]
     zone = parameters[self.PARAM_ZONE]
 
+    # In case of autoscaling, the server side passes these parameters as a
+    # string, so this check makes sure that spot instances are only created
+    # when the flag is True.
+    spot = False
+    if parameters[self.PARAM_SPOT] in ['True', 'true', True]:
+      spot = True
```

Comment:
3 lines could be replaced with:
`spot = parameters[self.PARAM_SPOT] in ['True', 'true', True]`